### PR TITLE
feat(toast): add stubbed swipeGesture

### PR DESCRIPTION
**Pull request by liamdebeasi**
Issue number: N/A

---------

## What is the current behavior?

Toast does not support swipe gestures to dismiss.

## What is the new behavior?

- Adds a `swipeGesture` property to enable this behavior
- Creates an HTML playground so I can later write E2E tests
- Adds stubbed implementations for creating and destroying the swipe gesture on prop change, present, and dismiss
- Adds unit tests to verify that the gesture is being created/destroyed on prop change, present, and dismiss

In https://github.com/ionic-team/ionic-framework/pull/28380/commits/cf7e8976a76ca338123abf3ddd6a3863c956f7db I moved the toast config tests to separate file. For some reason this was causing issues with my new tests. This might be a Stencil bug given that I got the following error:

```js
, getValue = (e, t) => getHostRef(e).$instanceValues$.get(t), setValue = (e, t, a, o) => {
                                     ^

TypeError: Cannot read properties of undefined (reading '$instanceValues$')
    at getValue (/Users/liamdebeasi/Ionic/ionic/core/node_modules/@stencil/core/internal/testing/index.js:508:38)
    at Toast.get [as position] (/Users/liamdebeasi/Ionic/ionic/core/node_modules/@stencil/core/internal/testing/index.js:538:13)
    at Toast.dismiss (/Users/liamdebeasi/Ionic/ionic/core/src/components/toast/toast.tsx:208:30)
    at runNextTicks (node:internal/process/task_queues:60:5)
    at listOnTimeout (node:internal/timers:533:9)
    at processTimers (node:internal/timers:507:7)
```

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Note: This PR is missing some build artifacts. I will add those once https://github.com/ionic-team/ionic-framework/pull/28379 is merged and synced into this branch.

### CODE REVIEW

File: core/src/components/toast/toast.tsx [hunk 1]
```diff
@@ -193,6 +196,41 @@ export class Toast implements ComponentInterface, OverlayInterface {
...
 
+  /**
+   * If set to 'vertical', the Toast can be dismissed with
+   * a swipe gesture. The swipe direction is determined by
+   * the value of the `position` property:
+   * `top`: The Toast can be swiped up to dismiss.
+   * `bottom`: The Toast can be swiped down to dismiss.
+   * `middle`: The Toast can be swiped up or down to dismiss.
+   */
+  @Prop() swipeGesture?: 'vertical';
```
Comment (sean-perkins) [inline core/src/components/toast/toast.tsx:207]: ~ I'm ok with this as-is, but any reason for not specifying the entire type as part of this PR? You'll have to update this again to add `top`, `bottom` and `middle` which will create an additional diff to review vs. being reviewed as part of this.

**Review by sean-perkins — APPROVED** [review]

**Review by thetaPC — COMMENTED** [review]
What's the recommended way to test this? I wasn't able to test it locally.

**Comment by liamdebeasi** [comment]
Right now this is all stubbed, so manually testing it isn't going to do much since there's no implementation associated with it. (i.e. you won't be able to do `swipeGesture="vertical"` and drag the toast since that's not implemented yet)

The idea behind a stub is to add just enough code to be compiled and show reviewers how this feature will be activated. In other words, this PR demonstrates the interface for the feature (props, types, methods, etc), but the implementation hasn't been added yet. In this particular case, I wanted to focus on the property setup, making it reactive, and calling the code that will create or destroy the gesture on present and dismiss. You could add some logging in `present` and `dismiss` to manually verify that `this.gesture` is being cleared, but I also wrote test that verify this.

**Review by thetaPC — APPROVED** [review]
LGTM

**Pull request merged**